src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,515 @@
+        public string ConfigFile
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ConfigFile));
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Nvidia code directory.
+        /// </summary>
+        protected string NvidiaDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, "mlperf", "NVIDIA");
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Pytorch code directory.
+        /// </summary>
+        protected string ExecutionDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.NvidiaDirectoryPath, "benchmarks", this.Model, "implementations", this.Implementation);
+            }
+        }
+
+        /// <summary>
+        /// The output directory of MLPerf.
+        /// </summary>
+        protected string OutputDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.ExecutionDirectoryPath, "results");
+            }
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the MLPerf Training workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.InitializationStarted", telemetryContext);
+
+            this.ThrowIfPlatformNotSupported();
+
+            await this.ThrowIfUnixDistroNotSupportedAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            MLPerfState state = await this.stateManager.GetStateAsync<MLPerfState>($"{nameof(MLPerfState)}", cancellationToken)
+                ?? new MLPerfState();
+
+            if (!state.Initialized)
+            {
+                // add user in docker group
+                await this.ExecuteCommandAsync("usermod", $"-aG docker {this.Username}", this.ExecutionDirectoryPath, cancellationToken);
+
+                // Setup Environment
+                await this.SetupEnvironmentAsync(cancellationToken);
+                state.Initialized = true;
+                await this.stateManager.SaveStateAsync<MLPerfState>($"{nameof(MLPerfState)}", state, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Creates setup for MLPerf Training workload.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected async Task SetupEnvironmentAsync(CancellationToken cancellationToken)
+        {
+            string dockerImageCommand = $"docker build --pull -t {this.GetContainerName()} .";
+            string dockerRunCommand = $"docker run --runtime=nvidia {this.GetContainerName()}";
+
+            await this.ExecuteCommandAsync(
```

Comment:
Please check other classes and use the new ExecuteCommand that takes in telemetry context so we are sending telemetry with every command